frontend/src/core/term/withTermsAndPlaceables.js

```diff
@@ -0,0 +1,52 @@
+/* @flow */
+
+import * as React from 'react';
+import escapeRegExp from 'lodash.escaperegexp';
+import createMarker from 'react-content-marker';
+import shortid from 'shortid';
+
+import {
+    getRulesWithFluent,
+    getRulesWithoutLeadingSpace,
+    rules,
+} from 'core/placeable';
+
+import type { TermState } from 'core/term';
+
+
+export default function withTermsAndPlaceables(terms: TermState, fluent: ?boolean) {
```

Comment:
I'd rename `fluent` into `usingFluent` or `isFluent`. Just "fluent" is a bit confusing.

Reply:
And I would rename this function as well. The word "with" made sense when it was an HOC, but now that it returns a Marker, I doesn't work as good. Maybe call it something like `getMarkerWithTermsAndPlaceables`? Maybe just `getMarker`, as it is in the `terms` namespace already.